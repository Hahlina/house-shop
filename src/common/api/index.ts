export * from './fetchApi';
export * from './services/propety';
