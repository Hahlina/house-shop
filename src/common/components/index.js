export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
