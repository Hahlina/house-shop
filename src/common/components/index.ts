export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
