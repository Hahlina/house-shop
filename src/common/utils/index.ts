export * from './convertPrice';
export * from './capitalizeFirstLetter';
