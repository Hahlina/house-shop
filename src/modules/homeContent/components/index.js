export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
