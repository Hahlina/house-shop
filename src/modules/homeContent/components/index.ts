export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
