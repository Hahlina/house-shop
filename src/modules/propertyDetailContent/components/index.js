export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
