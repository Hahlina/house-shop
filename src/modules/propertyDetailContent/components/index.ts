export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
