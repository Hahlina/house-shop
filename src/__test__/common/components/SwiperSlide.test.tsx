import { SwiperSlide } from 'common/components';

import { render, screen } from 'common/utils/test';

describe('SwiperSlide component', () => {
    const slidesChild = <p>Child</p>;

    it('should render without crashing', () => {
        render(<SwiperSlide />);
        expect(screen.queryByTestId('swiper-slide')).toBeInTheDocument();
    });

    it('should render children', () => {
        render(<SwiperSlide>{slidesChild}</SwiperSlide>);
        expect(screen.getByText(/child/i)).toBeInTheDocument();
    });

    it('should take a snapshot without children', () => {
        const { container } = render(<SwiperSlide />);
        expect(container).toMatchSnapshot();
    });

    it('should take a snapshot with children', () => {
        const { container } = render(<swiper-slide>{slidesChild}</swiper-slide>);
        expect(container).toMatchSnapshot();
    });
});
