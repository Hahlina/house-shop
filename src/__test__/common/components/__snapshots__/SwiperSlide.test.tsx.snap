// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwiperSlide component should take a snapshot with children 1`] = `
<div>
  <swiper-slide
    data-testid="swiper-slide"
  >
    <p>
      Child
    </p>
  </swiper-slide>
</div>
`;

exports[`SwiperSlide component should take a snapshot without children 1`] = `
<div>
  <swiper-slide
    data-testid="swiper-slide"
  />
</div>
`;
